analysis/src/macro_expansion.rs

```diff
@@ -33,22 +33,38 @@ impl<T: FieldElement> Folder<T> for MacroExpander<T> {
     type Error = Error;
 
     fn fold_machine(&mut self, mut machine: Machine<T>) -> Result<Machine<T>, Self::Error> {
-        machine.statements.iter_mut().for_each(|s| match s {
-            MachineStatement::InstructionDeclaration(_, _, Instruction { body, .. }) => {
-                match body {
-                    InstructionBody::Local(body) => {
-                        *body = self.expand_macros(std::mem::take(body))
-                    }
-                    InstructionBody::CallableRef(..) => {
+        machine.statements = machine
```

Comment:
This was changed to a functional approach, as a statement can yield many, and `iter_mut` does not allow that. Hence the usage of flat_map here.